src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkflowComponent.scala

```diff
@@ -451,45 +453,50 @@ trait WorkflowComponent {
       SubmissionValidationRecord(
         0,
         parentId,
         value.error,
-        value.inputName
+        attributeNamespaceId,
+        value.inputName.name
       )
     }
 
     //Unmarshal all input resolutions for a single workflow. Assumes that everything in wfInputResolutionRecs has the same workflowId.
-    def unmarshalOneWorkflowInputs(wfInputResolutionRecs: Seq[(SubmissionValidationRecord, Option[SubmissionAttributeRecord])], workflowId: Long): Seq[SubmissionValidationValue] = {
-
-        //collect up the workflow resolution results by input
-        val resolutionsByInput = wfInputResolutionRecs.groupBy { case (resolution, attribute) => resolution.inputName }
-
-        //unmarshalAttributes will unmarshal multiple workflow attributes at once, but it expects all the attribute records to be real and not options.
-        //To get around this, we split by input, so that each input is successful (or not) individually.
-        val submissionValues = resolutionsByInput map { case (inputName, recTuples: Seq[(SubmissionValidationRecord, Option[SubmissionAttributeRecord])]) =>
-          val attr = if( recTuples.forall { case (submissionRec, attrRecOpt) => attrRecOpt.isDefined } ) {
-            //all attributes are real
-            Some(
-              submissionAttributeQuery.unmarshalAttributes( recTuples map { case (rec, attrOpt) => ((workflowId, attrOpt.get), None) } )(workflowId)(inputName)
-            )
-          } else {
-            None
-          }
-          //assuming that the first elem has the error here
-          SubmissionValidationValue(attr, recTuples.head._1.errorText, inputName)
+    def unmarshalOneWorkflowInputs(namespaceMap: Map[String, Long], wfInputResolutionRecs: Seq[(SubmissionValidationRecord, Option[SubmissionAttributeRecord])], workflowId: Long): Seq[SubmissionValidationValue] = {
+
+      //collect up the workflow resolution results by input
+      val resolutionsByInput = wfInputResolutionRecs.groupBy { case (resolution, attribute) =>
+        val namespaceId = attributeNamespaceQuery.unmarshalNamespace(namespaceMap, resolution.inputNamespace)
+        AttributeName(namespaceId, resolution.inputName)
+      }
+
+      //unmarshalAttributes will unmarshal multiple workflow attributes at once, but it expects all the attribute records to be real and not options.
+      //To get around this, we split by input, so that each input is successful (or not) individually.
+      val submissionValues = resolutionsByInput map { case (inputName, recTuples: Seq[(SubmissionValidationRecord, Option[SubmissionAttributeRecord])]) =>
+        val attr = if (recTuples.forall { case (submissionRec, attrRecOpt) => attrRecOpt.isDefined }) {
+          //all attributes are real
+          val attrRecsWithRefs = recTuples map { case (rec, Some(attrOpt)) => ((workflowId, attrOpt), None) }
```

Comment:
This gives a "match may not be exhaustive" warning, but we know these are all defined based on the above check.